### ISSUE TITLE
where Hunter save (as default) Environment's key?

### ISSUE THREAD

**Issue by caprone** [HunterMcGushion/hyperparameter_hunter#87]
HI

where hunter save (as default) Environment's key?
because not in path specified in 'root_results_path' param;
then what if I need to reset or delete that Environment's key?
in fact if I rerun the same script (after deleting all files in "root_results_path" folder), many experiments are not saved in 'optimizer.similar_experiments', infact
"assert experiment.experiment_id in [_[2] for _ in optimizer.similar_experiments]"  raise Exception;

another question: why 'GlobalLeaderboard.csv' stores experiments in descending order also when using metrics like 'mean_squared_error'?
 then it report me the worse model as the best (with higher mean_squared_error score)...

thanks

**Comment by caprone** [comment]
HI and thanks for fast answer & great package.
well, It is very clear from the documentation use of 'root_results_path', no problem;
infact I setted a relative path to store records, and all works great;
for example I notice that "experiment id" are saved in /TestedKeys/ subdirectory;
I do not think it is necessary to make 'root_results_path' required argument, 
because sometimes it might be useful to explore a shrinked space on the fly for simple e fast tasks;
then displayed scores can be enough;


my problem is that it's not clear to me which experiments are saved in 'optimizer.similar_experiments';
when I run "functionality_example.py" script, not all experiment are 'saved in 'optimizer.similar_experiments';
but of course all of them are saved in 'GlobalLeaderboard.csv';
then most of the time "assert experiment.experiment_id in [_[2] for _ in optimizer.similar_experiments]" raise Expections..

**Comment by HunterMcGushion** [comment]
Great, I’m glad that’s cleared up and the documentation makes sense.

I see. Are you running the exact, unmodified, most recent `example/functionality_example` script? I just tried running it again, and everything went smoothly, so can you tell me a bit about your environment? OS? Python version? XGBoost version? HyperparameterHunter version? 

Thanks for your feedback, I’ll get to work on that unless you’ve got an idea and would like to submit a pull request?

Currently, the way to look at the varying hyperparameters and recorded scores for a set of experiments using the same algorithm involves reading the “GlobalLeaderboard.csv” to find the experiment ids you’re interested in, then opening the .json description files created for those experiments, in which you can find all hyperparameter and testing condition information. I hope to add algorithm-specific leaderboard files that will show hyperparameter values right on the leaderboard, but I haven’t gotten around to it yet.

**Comment by HunterMcGushion** [comment]
Great! Since the primary issue here now seems to be the need to tell the optimizer to minimize/maximize depending on the primary metric used, I'll close this issue as this is also mentioned in #34. If you disagree, please let me know, and we can reopen this! Thanks a lot for voicing your support for that feature!
